File: tp1/python/src/personne.py
class Personne:
    def __init__(self, nom, age):
        self.__nom = nom
        self.__age = age

    @property
    def nom(self):
        return self.__nom

    @property
    def age(self):
        return self.__age

    def afficher_details(self):
        print(f"Nom : {self.__nom}, Age : {self.__age}")

File: tp_architecture_logicielle_avancee/java/main/Personne.java
public abstract class Personne {
    protected String nom;
    protected int age;

    public Personne(String nom, int age) {
        setNom(nom);
        setAge(age);
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        if (nom == null || nom.trim().isEmpty()) {
            throw new IllegalArgumentException("Le nom ne peut pas etre vide");
        }
        this.nom = nom;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        if (age < 0 || age > 120) {
            throw new IllegalArgumentException("L'age doit etre compris entre 0 et 120");
        }
        this.age = age;
    }

    public abstract void afficherDetails();
}
